plugins/ptp_operator/ptp4lconf/ptp4lConfig.go

```diff
@@ -0,0 +1,233 @@
+	var interfaces []*string
+	if p.Ptp4lConf != nil {
+		matches := sectionHead.FindAllStringSubmatch(*p.Ptp4lConf, -1)
+		for _, v := range matches {
+			if v[1] != ptp4lGlobalSection && !checkIfPtP4lConf(v[1]) {
+				interfaces = append(interfaces, &v[1])
+			}
+		}
+	}
+	return interfaces
+}
+
+//PTPInterface ...
+type PTPInterface struct {
+	Name     string
+	PortID   int
+	PortName string
+	Role     types.PtpPortRole
+}
+
+//PTP4lConfig ...
+type PTP4lConfig struct {
+	Name       string
+	Interfaces []*PTPInterface
+}
+
+//ByRole ...
+func (Ptp4lCfg *PTP4lConfig) ByRole(role types.PtpPortRole) (PTPInterface, error) {
+	for _, p := range Ptp4lCfg.Interfaces {
+		if p != nil && p.Role == role {
+			return *p, nil
+		}
+	}
+	return PTPInterface{}, fmt.Errorf("interfaces not found for the role %d --> %s", role, Ptp4lCfg.String())
+}
+
+//UpdateRole ...
+func (pi *PTPInterface) UpdateRole(role types.PtpPortRole) {
+	pi.Role = role
+}
+
+func (Ptp4lCfg *PTP4lConfig) String() string {
+	b := strings.Builder{}
+	b.WriteString("  configName: " + Ptp4lCfg.Name + "\r\n")
+	b.WriteString("  Ptp4lConfigInterfaces: " + "\n")
+	for _, p := range Ptp4lCfg.Interfaces {
+		b.WriteString("  name: " + p.Name + "\n")
+		b.WriteString("  portName: " + p.PortName + "\n")
+		b.WriteString("  role: " + p.Role.String() + "\n")
+		b.WriteString("  portID: " + fmt.Sprintf("%d", p.PortID) + "\n")
+	}
+	return b.String()
+}
+
+//ByInterface ...
+func (Ptp4lCfg *PTP4lConfig) ByInterface(iface string) (PTPInterface, error) {
+	for _, p := range Ptp4lCfg.Interfaces {
+		if p != nil && p.Name == iface {
+			return *p, nil
+		}
+	}
+	return PTPInterface{}, fmt.Errorf("interfaces not found for the interface %s", iface)
+}
+
+//ByPortID ...
+func (Ptp4lCfg *PTP4lConfig) ByPortID(id int) (PTPInterface, error) {
+	for _, p := range Ptp4lCfg.Interfaces {
+		if p != nil && p.PortID == id {
+			return *p, nil
+		}
+	}
+	return PTPInterface{}, fmt.Errorf("interface was not found for the port id %d", id)
+}
+
+// Watcher monitors a file for changes
+type Watcher struct {
+	fsWatcher *fsnotify.Watcher
+	close     chan struct{}
+}
+
+//Close ...
+func (w *Watcher) Close() {
+	close(w.close)
+}
+
+//NewPtp4lConfigWatcher ...
+func NewPtp4lConfigWatcher(dirToWatch string, updatedConfig chan<- *PtpConfigUpdate) (w *Watcher, err error) {
+
+	fsWatcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Errorf("error setting for file change notifier %s", err)
+	}
+
+	w = &Watcher{
+		fsWatcher: fsWatcher,
+		close:     make(chan struct{}),
+	}
+
+	go func() {
+		defer w.fsWatcher.Close()
+		//initialize all
+		for _, ptpConfig := range readAllConfig(dirToWatch) {
+			log.Infof("sending ptpconfig changes for %s", *ptpConfig.Name)
+			updatedConfig <- ptpConfig
+		}
+		for {
+			select {
+			case event, ok := <-w.fsWatcher.Events:
+				if !ok {
+					log.Error("could not read watcher events")
+					continue
+				}
+				log.Infof("event: %v", event)
```

Comment:
are there a lot of events? if so might want to change to debugf.

Reply:
this event is only when you modify(add/remove)   ptp4l.x.config file, I will removed the info ,Which is not required